lib/ziti_tunnel_ctrl.c

```diff
@@ -0,0 +1,245 @@
+
+static struct cmd_ctx_s {
+    ziti_tunnel_ctrl ctrl;
+    tunneler_context tunnel_ctx;
+    command_cb cb;
+    uv_loop_t *loop;
+} CMD_CTX;
+
+struct ziti_instance_s {
+    ziti_options opts;
+    command_cb load_cb;
+    void *load_ctx;
+
+    ziti_context ztx;
+    LIST_ENTRY(ziti_instance_s) _next;
+};
+
+// temporary list to pass info between parse and run
+static LIST_HEAD(instance_list, ziti_instance_s) instance_init_list;
+
+// map<path -> ziti_instance>
+static model_map instances;
+
+static void on_ziti_event(ziti_context ztx, const ziti_event_t *event);
+
+static const char * cfg_types[] = { "ziti-tunneler-client.v1", "intercept.v1", "ziti-tunneler-server.v1", "host.v1", NULL };
+
+static long refresh_interval = 10;
+
+static int process_cmd(const tunnel_comand *cmd, void (*cb)(const tunnel_result *, void *ctx), void *ctx);
+static int load_identity(const char *path, command_cb cb, void *ctx);
+static struct ziti_instance_s *new_ziti_instance(const char *path);
+static void load_ziti_async(uv_async_t *ar);
+
+const ziti_tunnel_ctrl* ziti_tunnel_init_cmd(uv_loop_t *loop, tunneler_context tunnel_ctx, command_cb cb) {
+    CMD_CTX.loop = loop;
+    CMD_CTX.tunnel_ctx = tunnel_ctx;
+    CMD_CTX.cb = cb;
+    CMD_CTX.ctrl.process = process_cmd;
+    CMD_CTX.ctrl.load_identity = load_identity;
+    return &CMD_CTX.ctrl;
+}
+
+static int process_cmd(const tunnel_comand *cmd, command_cb cb, void *ctx) {
+    tunnel_result result = {
+                    .success = false,
+                    .error = NULL,
+                    .data = NULL,
+    };
+    ZITI_LOG(INFO, "processing command[%s] with data[%s]", TunnelCommands.name(cmd->command), cmd->data);
+    switch (cmd->command) {
+        case TunnelCommand_LoadIdentity: {
+            load_identity_cmd load;
+            if (cmd->data == NULL || parse_load_identity_cmd(&load, cmd->data, strlen(cmd->data)) != 0) {
+                result.success = false;
+                result.error = "invalid command";
+                break;
+            }
+            load_identity(load.path, cb, ctx);
+            return 0;
+        }
+
+        case TunnelCommand_ListIdentities: {
+            identity_list id_list = {0};
+            id_list.identities = calloc(model_map_size(&instances) + 1, sizeof(identity_info*));
+            const char *key;
+            struct ziti_instance_s *inst;
+            int idx = 0;
+            MODEL_MAP_FOREACH(key, inst, &instances) {
+                identity_info *info = alloc_identity_info();
+                const ziti_identity *identity = ziti_get_identity(inst->ztx);
+                info->name = strdup(identity->name);
+                info->id = strdup(identity->id);
+                info->network = strdup(ziti_get_controller(inst->ztx));
+                info->config = strdup(key);
+
+                id_list.identities[idx++] = info;
+            }
+
+            result.data = identity_list_to_json(&id_list, MODEL_JSON_COMPACT, NULL);
```

Comment:
is this the right place to set `result.success` to true?

currently seeing this:
```
$ echo -n '{ "command":"ListIdentities" }' | ncat -U /tmp/ziti-edge-tunnel.sock 
{"success":false,"data":{"identities":[{"name":"scarey@fedora-33-vm","config":"/media/psf/Home/.ziti/scarey@fedora-33-vm.json","network":"https://lambda-caeli.localdomain:11281","id":"jUSKpnLSx"},{"name":"mattermost.shawn.carey.pc","config":"/media/psf/Home/.ziti/mattermost.shawn.carey.pc_Enrollment.json","network":"https://3.210.34.143:443","id":"9pv6IQQYC"}]}}
```